### PR TITLE
Export unified ParachainHostFunctions

### PR DESCRIPTION
**Pull request by skunert**
This PR exports unified hostfunctions needed for parachains. Basicaly `SubstrateHostFunctions` + `storage_proof_size::HostFunctions`.

Also removes the native executor from the parachain template.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

File: cumulus/client/service/src/lib.rs [hunk 1]
```diff
@@ -54,6 +54,15 @@ use std::{sync::Arc, time::Duration};
 
 pub use cumulus_primitives_proof_size_hostfunction::storage_proof_size;
 
+/// Hostfunctions that should be used in parachain nodes.
```
Comment (michalkucharczyk) [inline cumulus/client/service/src/lib.rs:57]:
```suggestion
/// Host functions that should be used in parachain nodes.
```

File: cumulus/client/service/src/lib.rs [hunk 2]
```diff
@@ -54,6 +54,15 @@ use std::{sync::Arc, time::Duration};
...
+///
+/// Contains the standard substrate hostfunctions, as well as a
```
Comment (michalkucharczyk) [inline cumulus/client/service/src/lib.rs:59]:
```suggestion
/// Contains the standard substrate host functions, as well as a
```

File: cumulus/client/service/src/lib.rs [hunk 3]
```diff
@@ -54,6 +54,15 @@ use std::{sync::Arc, time::Duration};
...
+/// hostfunction to enable PoV-reclaim on parachain nodes.
```
Comment (michalkucharczyk) [inline cumulus/client/service/src/lib.rs:60]:
```suggestion
/// host function to enable PoV-reclaim on parachain nodes.
```

**Review by michalkucharczyk — APPROVED** [review]

**Comment by michalkucharczyk** [comment]
left some nits: the phrase "host function" is more often used in the comments across the codebase.

**Pull request merged**